### PR TITLE
Migrating mimikatz.dll to pypykatz

### LINKED ISSUE #583: Replace Mimikatz with pypykatz for better defence evasion
**Is your feature request related to a problem? Please describe.**
Most AVs recognize and delete the Mimikatz DLL when we unzip it to disk and therefore we don't steal creds

**Describe the solution you'd like**
Moving to Pykatz will circumvent this defence mechanism.

**Describe alternatives you've considered**
Different loaders for the DLL, obfuscation of the DLL itself

### PR DESCRIPTION
**Pull request by EddieIvan01**
# What is this? 
Fixes #583, for GSoC proposal

Add any further explanations here. 

## Checklist
* [x] Migrating mimikatz.dll to pypykatz

## Proof that it works
The `MimikatzCollector.get_logon_info` output on my machine ( Windows 10 ), it could dump the NTLM hash.
```
{'NEMESIS$': {}, 'UMFD-0': {}, 'LOCAL SERVICE': {}, 'Eddie_Ivan': {'ntlm_hash': '4f9099ea6ffad7906207dbc841ef13d4'}, 'UMFD-14': {}, 'DWM-14': {}}
```
May need to test on more OS.

### CODE REVIEW

File: monkey/infection_monkey/system_info/mimikatz_collector.py [hunk 1]
```diff
@@ -58,6 +53,75 @@ def init_mimikatz(self):
         T1106Telem(status, UsageEnum.MIMIKATZ_WINAPI).send()
         T1129Telem(status, UsageEnum.MIMIKATZ).send()
 
+    def _init_pypykatz(self):
+        from pypykatz.pypykatz import pypykatz
```
Comment (danielguardicore) [inline monkey/infection_monkey/system_info/mimikatz_collector.py]: Need to update requirements.txt

**Comment by danielguardicore** [comment]
Nice work, will review properly and add comments.
Why did you decide to keep the mimikatz?

File: monkey/infection_monkey/system_info/mimikatz_collector.py (hunk 1 above)
Comment (EddieIvan01) [reply monkey/infection_monkey/system_info/mimikatz_collector.py]: OK

**Comment by ShayNehmad** [comment]
Hi @EddieIvan01 please talk with me via Slack (GSoC slack or Infection Monkey Slack are both fine)  about GSoC - you need to submit a proposal and I'd like to talk with you beforehand :)

**Comment by EddieIvan01** [comment]
@danielguardicore I think mimikatz may still be useful at some point (perhaps when pypykatz can't work well, it's still version beta 0.1.1). And I think the mimikatz.dll should be updated version(it's 1.1 now)

**Comment by EddieIvan01** [comment]
@ShayNehmad, yeah. I have submitted my proposal on the official website of GSoC yesterday. I'll talk with you via Infection Monkey Slack :)

**Comment by acepace** [comment]
@EddieIvan01 sorry to get to this so late.
On what version of Windows 10 did this work for you?

**Comment by EddieIvan01** [comment]
Windows10 1903

**Pull request closed without merging**